coordinator/src/node.rs

```diff
@@ -208,17 +192,45 @@ enum TradeAction {
     // Reduce,
 }
 
+/// Calculates the accept settlement amount based on the pnl.
+///
+/// The smallest possible result is 0.
+fn calculate_accept_settlement_amount(
+    opening_price: Decimal,
+    closing_price: Decimal,
+    quantity: f64,
+    long_leverage: f64,
+    short_leverage: f64,
+    direction: Direction,
+) -> Result<u64> {
+    let pnl = cfd::calculate_pnl(
+        opening_price,
+        closing_price,
+        quantity,
+        long_leverage,
+        short_leverage,
+        direction,
+    )?;
+
+    let margin_trader = calculate_margin(opening_price, quantity, long_leverage);
+
+    let accept_settlement_amount = (margin_trader as i64).saturating_add(pnl);
+    // the amount can only be positive, adding a safeguard here with the max comparison to
+    // ensure the i64 fits into u64
+    Ok(accept_settlement_amount.max(0) as u64)
```

Comment:
👍 As a small nit, I would prefer if an important operation was on a line other than the return position. Otherwise it's too easy to ignore.